frontend/src/app/main/ui/workspace/viewport/outline.cljs

```diff
@@ -90,18 +91,17 @@
         show-outline? (fn [shape] (and (not (:hidden shape))
                                        (not (:blocked shape))))
 
-        shapes (set
-                (into
-                 (->> outlines-ids
-                      (filter #(not= edition %))
-                      (map #(get objects %))
-                      (filterv show-outline?)
-                      (filter some?))
-                 ;; outline highlighted shapes even if they are hidden or blocked
-                 (->> highlighted
-                      (filter #(not= edition %))
-                      (map #(get objects %))
-                      (filter some?))))]
+        shapes (d/concat-set
```

Comment:
That's a nice helper! Surprised it's not called 'setcat' :D

Looks more declarative / semantical

Reply:
Hmm, I think you are pretty right on this! Now that you have mentioned the name, it looks more interesting that the current one hehe. I will propose/do the change the next week If I have "free" time hehe.